graylog2-server/src/main/java/org/graylog2/indexer/retention/strategies/AbstractIndexCountBasedRetentionStrategy.java

```diff
@@ -89,19 +90,29 @@ private void runRetention(IndexSet indexSet, Map<String, Set<String>> deflectorI
             .filter(indexName -> !(deflectorIndices.getOrDefault(indexName, Collections.emptySet()).contains(indexSet.getWriteIndexAlias())))
             .sorted((indexName1, indexName2) -> indexSet.extractIndexNumber(indexName2).orElse(0).compareTo(indexSet.extractIndexNumber(indexName1).orElse(0)))
             .collect(Collectors.toCollection(LinkedHashSet::new));
+
+        LinkedList<String> orderedIndicesDescending = new LinkedList<String>();
+
         orderedIndices
-            .stream()
-            .skip(orderedIndices.size() - removeCount)
-             // reverse order to archive oldest index first
-            .collect(Collectors.toCollection(LinkedList::new)).descendingIterator()
-            .forEachRemaining(indexName -> {
-                final String strategyName = this.getClass().getCanonicalName();
-                final String msg = "Running retention strategy [" + strategyName + "] for index <" + indexName + ">";
-                LOG.info(msg);
-                activityWriter.write(new Activity(msg, IndexRetentionThread.class));
-
-                // Sorry if this should ever go mad. Run retention strategy!
-                retain(indexName, indexSet);
-            });
+                .stream()
+                .skip(orderedIndices.size() - removeCount)
+                // reverse order to archive oldest index first
+                .collect(Collectors.toCollection(LinkedList::new)).descendingIterator().
+                forEachRemaining(indexName -> orderedIndicesDescending.add(indexName));
+
+        String indexNames = "";
+        for(String index : orderedIndicesDescending) {
+            if(!indexNames.isEmpty()){
+                indexNames.concat(",");
```

Comment:
This doesn't work. You'd need to assign the result to the variable again.
However, there is a much simpler way of doing this:
`indexNames = orderedIndicesDescending.stream().collect(Collectors.joining(","));`